packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1696,6 +1696,60 @@ abstract class ParentDataWidget<T extends ParentData> extends ProxyWidget {
 /// for that inherited widget using [BuildContext.dependOnInheritedWidgetOfExactType]
 /// and then returns the [ThemeData].
 ///
+/// When using the `of` method, the `context` must be a descendant of the
+/// [InheritedWidget], meaning it must be "bellow" the [InheritedWidget] in the
+/// tree.
+///
+/// {@tool snippet}
+///
+/// DO
+///
+/// ```dart
+/// class MyPage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: FrogColor(
+///         color: Colors.green,
+///         child: Builder(
+///           builder: (builderContext) {
+///             return Text(
+///               'Hello Frog',
+///               // note how `builderContext` is bellow `FrogColor`
```

Comment:
below

Reply:
removed the comments entirely as we now have similar text before each snippet